### ISSUE TITLE
[HOLD for payment 2024-11-28] [$250] Attachment - The position of the attachment "Menu" modal is incorrect

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#50818]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.49-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to https://staging.new.expensify.com/
2. Log in with any account
3. Navigate to any 1:1 DM
4. Send any video attachment
5. Send any image attachment
6. Click on the video to play it
7. Hover over the video and click on the "Fullscreen" button
8. Minimize the video
9. Hover over the image attachment and click on the "..." button
10. Hover over the video attachment and click on the "..." button

## Expected Result:
"Menu" modal should appear close the attachment, but not overlapping any elements

## Actual Result:
The position of the attachment "Menu" modal is incorrect. Sometimes it appears far away from the attachment, sometimes it overlaps the quick menu buttons and sometimes it appears on the LHN

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/d831f671-2491-4f6c-8a70-363883be190d

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021846647666370282624</li>
        <li>Upwork Job ID: 1846647666370282624</li>
        <li>Last Price Increase: 2024-10-16</li>
<li>Automatic offers: </li>
<ul>
<li>rojiphil | Reviewer | 104546765</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @RachCHopkins</details>

**Comment by siddhakdak** [comment]
@RachCHopkins  can i start work on this issue.

**Comment by siddhakdak** [comment]
Contributor details
Your Expensify account email: siddhakdak19@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01e9eca845d1af9067

**Comment by RachCHopkins** [comment]
I can't reproduce this - it's exactly where it should be every time for me. I'm on v9.0.49-1 so maybe it's been fixed organically.

Can we get a retest, please?

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by RachCHopkins** [comment]
@siddhakdak The process is for the issue to be triaged before any assignment. Right now we're still working out if it's a problem. If you can reproduce the issue, then that can help move things along towards the proposal stage. Otherwise I'm asking our testers to check anyhow.

**Comment by FitseTLT** [comment]
BTW @RachCHopkins I easily reproduced it following the steps, are U interested in fixing it

https://github.com/user-attachments/assets/4ff4d0b7-6cb9-48da-aee2-92be0a7a1dd5

**Comment by izarutskaya** [comment]
Issue is still reproducible 
Build v9.0.49-1

https://github.com/user-attachments/assets/b0428d9c-2022-46c7-a5cb-b87390dee15b

**Comment by RachCHopkins** [comment]
Thanks everyone, I think we can confirm it needs fixing now!

**Comment by teneeto** [comment]
Hey! I'm Eto Olei from Callstack. I will like to take a look at this issue. Please assign it to me. Thanks!

**Comment by shahinyan11** [comment]
Hi there . I has Help Wanted label and I am working on it already

**Comment by teneeto** [comment]
Oops! @shahinyan11 go for it then.

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Attachment - The position of the attachment "Menu" modal is incorrect
### What is the root cause of that problem?
[Here](https://github.com/Expensify/App/blob/987ff1b536f2144e5fd492c6d2d2aa621d68442f/src/hooks/useWindowDimensions/index.ts#L124C1-L137C6) we have a logic to lock and unlock window dimensions. And we should unlock  dimensions when user exit fullscreen mode as described [here](https://github.com/Expensify/App/blob/987ff1b536f2144e5fd492c6d2d2aa621d68442f/src/hooks/useWindowDimensions/index.ts#L132C5-L132C68). But sometimes after exiting fullscreen mode the unlock logic is not called because `didScreenReturnToOriginalSize` is `false`.  
The `didScreenReturnToOriginalSize` sometimes  is `false` because it is the result of the comparison if the `lockedWindowDimensionsRef` dimensions are equal to the current window dimensions and we set `lockedWindowDimensionsRef` after that the `isFullScreenRef.current` becomes true. But sometimes  the activating fullscreen mode works before the setting of `lockedWindowDimensionsRef` and sometimes after it. When activation fullscreen works faster then we set the full screen dimensions to `lockedWindowDimensionsRef` and after exiting fullscreen mode the current window dimensions will not be equal to `lockedWindowDimensionsRef`


### What changes do you think we should make in order to solve the problem?
The existence of  `didScreenReturnToOriginalSize` in [this](https://github.com/Expensify/App/blob/987ff1b536f2144e5fd492c6d2d2aa621d68442f/src/hooks/useWindowDimensions/index.ts#L133C9-L133C103) condition does not ensure that the code will always work after exiting fullscreen mode. So that can remove it from condition
```
if (lockedWindowDimensionsRef.current && !isFullScreenRef.current ) {
```


### What alternative solutions did you explore? (Optional)
We can call `unlockWindowDimensions()` in [this](https://github.com/Expensify/App/blob/62302d5cacb2ff307b2e3a2bfc267a858a79bb4d/src/components/VideoPlayer/BaseVideoPlayer.tsx#L238) line to ensure that it will be called on exiting fullscreen mode

**Comment by RachCHopkins** [comment]
@shahinyan11 the Help Wanted label means we are inviting proposals, there is no guarantee your proposal will be accepted. You cannot lay "claim" to an issue like this. Please refamiliarise yourself with the [Contributor Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md).

@teneeto works for an agency contracted to us, and is welcome to take on the job right up until a proposal is accepted, without first submitting a proposal.

**Comment by RachCHopkins** [comment]
@shahinyan11 the Help Wanted label means we are inviting proposals, there is no guarantee your proposal will be accepted. You cannot lay "claim" to an issue like this. Please refamiliarise yourself with the [Contributor Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md).

@teneeto works for an agency contracted to us, and is welcome to take on the job right up until a proposal is accepted, without first submitting a proposal.

**Comment by shahinyan11** [comment]
@RachCHopkins Sorry but I do not agree with you. I ready the  [Contributor Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) and understand that the "Help Wonted" label gives a guarantee for following -  `if for resolving issue will use your solution you should be paid`.  If not then the contribution is not guaranteed that his proposal will not be used without compensation. Sorry if I am wrong

**Comment by teneeto** [comment]
Hi @RachCHopkins @rojiphil, I got halted on this topic to investigate another issue, but I am checking this now. I will drop updates as I move forward.

**Comment by RachCHopkins** [comment]
Thanks @teneeto, looking forward to it!

**Comment by RachCHopkins** [comment]
@shahinyan11 I don't think you entirely understand the established process. Please take this to the Slack channel if you want to discuss it, as this issue is not really the place for this conversation! 😄

**Comment by teneeto** [comment]
Hey, I'm able to reproduce this, and it certainly has to do with the window dimensions. I had initially come up with a solution, but then i noticed it might not be the best. I'm trying an alternative and will continue with updates here.

**Comment by RachCHopkins** [comment]
How's the alternative plan going @teneeto ?

**Comment by teneeto** [comment]
Hi @RachCHopkins looking at raising PR today. should be good on that soon.

**Comment by teneeto** [comment]
Hi @RachCHopkins, a PR is ready, let's take a look.

**Comment by RachCHopkins** [comment]
@rojiphil I'm just waiting on your checklist here before I issue payment.

**Comment by rojiphil** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production
  - [ ] 2b. Reported on staging (deploy blocker)
  - [ ] 2c. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [x] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    [Link to comment](https://github.com/Expensify/App/pull/38407/files#r1863542513): 

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: Not Required. The existing checklist is good enough to capture such issues.

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

## Regression Test Proposal
### Precondition:

### Test:
1. Navigate to any 1:1 DM
2. Send any video attachment
3. Send any image attachment
4. Click on the video to play it
5. Hover over the video and click on the "Fullscreen" button
6. Minimize the video
7. Hover over the image attachment and click on the "..." button
8. Hover over the video attachment and click on the "..." button
9. Verify that the popover menu position is correct with respect to the report actions in steps 7 and 8

Do we agree 👍 or 👎

**Comment by rojiphil** [comment]
> I'm just waiting on your checklist here before I issue payment.

@RachCHopkins Sorry for the delay here. I have completed the BZ checklist [here](https://github.com/Expensify/App/issues/50818#issuecomment-2507846011). Thanks.

**Comment by RachCHopkins** [comment]
Payment Summary:

- Contributor: @teneeto paid via contract
- Contributor+: @rojiphil paid $250 via Upwork

Upwork job [here](https://www.upwork.com/jobs/~021846647666370282624)

**Comment by RachCHopkins** [comment]
Contributor has been paid, the contract has been completed, and the Upwork post has been closed.
